src/meta/src/rpc/ddl_controller.rs

```diff
@@ -2172,6 +2356,14 @@ impl DdlController {
             old_fragments.max_parallelism,
         );
 
+        if let Some(sinks) = &auto_refresh_schema_sinks {
+            for sink in sinks {
+                replace_upstream
+                    .remove(&sink.new_fragment.fragment_id)
+                    .expect("should exist");
```

Comment:
Why do we need to remove the sink fragment?

Reply:
The replace_upstream is used for normal alter table. 

For example, let's say we have a table, and a downstream mv or non-auto-schema change sink created on the table. When we alter table and add a column, the fragment of the table will be changed to a new fragment, and then the existing fragments of downstream jobs should do a `replace_upstream`, to replace its original upstream fragment id to the table fragment to the new table fragment. However, for sinks with auto schema change, its existing fragment will be dropped, and new fragment will be created, and therefore there won't be a `replace_upstream` happened for the previous sink fragment, and then we should remove it here.